awx/main/scheduler/task_manager.py

```diff
@@ -484,15 +513,22 @@ def process_pending_tasks(self, pending_tasks):
                     break
 
                 if idle_instance_that_fits is None:
-                    idle_instance_that_fits = rampart_group.find_largest_idle_instance()
+                    idle_instance_that_fits = rampart_group.find_largest_idle_instance(self.graph[rampart_group.name]['instances'])
+                    if idle_instance_that_fits:
+                        idle_instance_that_fits.capacity -= task.task_impact
+                        idle_instance_that_fits.jobs_running += 1
+                        idle_instance_that_fits = idle_instance_that_fits.obj
                 remaining_capacity = self.get_remaining_capacity(rampart_group.name)
                 if not rampart_group.is_containerized and self.get_remaining_capacity(rampart_group.name) <= 0:
                     logger.debug("Skipping group {}, remaining_capacity {} <= 0".format(
                                  rampart_group.name, remaining_capacity))
                     continue
 
-                execution_instance = rampart_group.fit_task_to_most_remaining_capacity_instance(task)
+                execution_instance = rampart_group.fit_task_to_most_remaining_capacity_instance(task, self.graph[rampart_group.name]['instances'])
                 if execution_instance:
+                    execution_instance.capacity -= task.task_impact
```

Comment:
could we be double subtracting here if `idle_instance_that_fits == execution_instance`?